.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,55 @@
+{
+  "name": "io-services-cms",
+  "image": "mcr.microsoft.com/devcontainers/base:debian@sha256:6155a486f236fd5127b76af33086029d64f64cf49dd504accb6e5f949098eb7e",
+  "features": {
+    "ghcr.io/devcontainers-contrib/features/pre-commit:2": {},
+    "ghcr.io/devcontainers/features/azure-cli:1": {},
+    "ghcr.io/devcontainers/features/github-cli:1": {},
+    "ghcr.io/devcontainers/features/terraform:1": {
+      "version": "1.10.4",
+      "tflint": "latest",
+      "installTFsec": true,
+      "installTerraformDocs": true
+    },
+    "ghcr.io/pagopa/devcontainer-features/trivy:1": {},
+    "ghcr.io/pagopa/devcontainer-features/plantuml:1": {},
+    "ghcr.io/devcontainers/features/node:1": {
+      "version": "20.19.0",
+      "installYarnUsingApt": false,
+      "pnpmVersion": "none",
+      "nvmInstallPath": "/opt/nvm"
+    }
+  },
+  "postStartCommand": {
+    "set-workspace-as-safe-dir": "git config --global --add safe.directory ${containerWorkspaceFolder}",
+    "install-yarn-and-deps": "COREPACK_ENABLE_DOWNLOAD_PROMPT=0 yarn install"
+  },
+  "customizations": {
+    "vscode": {
+      "extensions": [
+        "-vscjava.vscode-java-pack",
+        "-ms-dotnettools.csharp",
+        "-ms-azuretools.vscode-azureterraform",
+        "ms-azuretools.vscode-azure-github-copilot",
+        "ms-vscode.vscode-websearchforcopilot",
+        "azurite.azurite",
+        "azapi-vscode.azapi",
+        "github.vscode-github-actions",
+        "github.vscode-pull-request-github",
+        "arcanis.vscode-zipfs",
+        "esbenp.prettier-vscode",
+        "EditorConfig.EditorConfig",
+        "bruno-api-client.bruno",
+        "vitest.explorer",
+        "redhat.vscode-yaml",
+        "dbaeumer.vscode-eslint",
+        "github.copilot",
+        "github.copilot-chat"
+      ],
+      "settings": {
+        "editor.formatOnSave": true,
+        "editor.defaultFormatter": "esbenp.prettier-vscode"
+      }
```

Comment:
I suggest to move these settings to `.vscode/settings.json`, to make them also available for those who don't use devcontainers

Don't forget to configure also the `defaultFormatter` for terraform files (https://github.com/pagopa/io-messages/blob/main/.vscode/settings.json#L5), the extension needed is included automatically by the `ghcr.io/devcontainers/features/terraform:1` feature

If you need `func`, add our feature https://github.com/pagopa/dx/tree/main/.devcontainer/features/azure-functions-core-tools instead of the official one, that is not compatible with apple silicon

Reply:
With the addition of `azure-functions-core-tools` feature, node scripts using `func` fail due to the presence of `“azure-functions-core-tools”:“^4.0.7030”` as the project's devDependency, which takes precedence over the globally installed one.
We introduced it as devDependency to centralize the dependency with a self-contained cross-platform package....do you know a way to "force" the container to use the system globally installation instead of the one with devDependency? Otherwise I think we should remove that devDependency in order to use DevContainer, correct?

Reply:
Just remove that `dev dependency`, the best way to install `func` is by `brew` on macOS and system PM on linux.

Even though it might seem convenient to include it among those dependencies, it not only creates "priority" issues within the system, but it also significantly slows down our CI/CD workflows, as it's a very heavy dependency.